les/vflux/server/prioritypool.go

```diff
@@ -311,40 +307,55 @@ func (pp *priorityPool) connectedNode(c *ppNodeInfo) {
 func (pp *priorityPool) disconnectedNode(c *ppNodeInfo) {
 	pp.lock.Lock()
 	pp.activeQueue.Refresh()
-	var updates []capUpdate
-	defer func() {
-		pp.lock.Unlock()
-		pp.updateFlags(updates)
-	}()
-
 	if !c.connected {
+		pp.lock.Unlock()
 		return
 	}
 	c.connected = false
 	pp.activeQueue.Remove(c.activeIndex)
 	pp.inactiveQueue.Remove(c.inactiveIndex)
+
+	var updates []capUpdate
 	if c.capacity != 0 {
-		pp.setTempState(c)
+		pp.setTempState(c, pp.minCap, pp.capacityStepDiv, 0)
 		pp.setTempCapacity(c, 0)
 		updates = pp.tryActivate(true)
 	}
+	pp.lock.Unlock()
+	pp.updateFlags(updates)
 }
 
 // setTempState internally puts a node in a temporary state that can either be reverted
 // or confirmed later. This temporary state allows changing the capacity of a node and
 // moving it between the active and inactive queue. activeFlag/inactiveFlag and
 // capacityField are not changed while the changes are still temporary.
-func (pp *priorityPool) setTempState(c *ppNodeInfo) {
+func (pp *priorityPool) setTempState(c *ppNodeInfo, minTarget uint64, stepDiv uint64, bias time.Duration) {
```

Comment:
It's a nice thing that temp parameter changes are explicitly tied to the temp state but I'm not sure that they really need to be set in the same function that enters the temp state. In many cases they are not changed at all, and in `tryActivate` I think they might need to change again (set bias to zero) while still in temp state (see my comment there). So maybe it'd be better to have a separate function for entering the temp state and another one for changing the temp state parameters (which could just do a check that we are really in temp state). Just like `setTempCapacity` that can change `tempCapacity` any time while in temp state.